Source/MoreDotNet/Extensions/Common/ByteArrayExtensions.cs

```diff
@@ -33,18 +33,24 @@ public static string GetString(this byte[] buffer)
             {
                 encoding = Encoding.UTF8;
             }
-            else if (buffer[0] == 0xfe && buffer[1] == 0xff)
+
+            // In addition to preamble check the length to help UTF16 with leading zeros be recognized properly as UTF32 is 4 bytes fixed with and UTF 16 is 2
```

Comment:
Small typos:
`fixed with` -> `fixed width`
`UTF16` -> `UTF-16`
`UTF 16` -> `UTF-16`
`UTF32` -> `UTF-32`

Also UTF-16 is has variable width of 2 or 4 bytes. It's good to update the comment. 😸